test/promql/tests/test_crashloopbackoff.yaml

```
@@ -0,0 +1,63 @@
+evaluation_interval: 1m
+
+rule_files:
+  - ../extracted-rules.yaml
+
+tests:
+  - interval: 1m
+    input_series:
+
+      # Pod failing with ImagePullBackOff error so it's not alerted
+      - series: 'kube_pod_container_status_waiting_reason{namespace="prod", pod="prod-pod-1", container="test-container", reason="ImagePullBackOff"}'
+        values: '1x15'
+
+      # Pod failing with CrashLoopBackOff error so it's alerted
+      - series: 'kube_pod_container_status_waiting_reason{namespace="test-ns", pod="prod-pod-2", container="test-container", reason="CrashLoopBackOff"}'
+        values: '1x15'
+
+      # Pod failing with CrashLoopBackOff error but it was resetted it between so this is not alerted
+      - series: 'kube_pod_container_status_waiting_reason{namespace="prod-test", pod="prod-pod-3", container="test-container", reason="CrashLoopBackOff"}'
+        values: '1x5 0x5 1x5'
+
+      # Pod failing with CrashLoopBackOff error for first 10 minutes so it's alerted
+      - series: 'kube_pod_container_status_waiting_reason{namespace="prod-ns", pod="prod-pod-4", container="test-container", reason="CrashLoopBackOff"}'
+        values: '1x10 0x5'
+
+      # Pod failing with CrashLoopBackOff error but the namespace ends with tenant so it's ignored
+      - series: 'kube_pod_container_status_waiting_reason{namespace="prod-tenant", pod="prod-pod-3", container="test-container", reason="CrashLoopBackOff"}'
```

```suggestion
      - series: 'kube_pod_container_status_waiting_reason{namespace="prod-tenant", pod="prod-pod-5", container="test-container", reason="CrashLoopBackOff"}'
```

Because `prod-pod-3` is already in use in another series.